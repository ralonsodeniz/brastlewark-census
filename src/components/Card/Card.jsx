import React from 'react';
import PropTypes from 'prop-types';

import {
  CardContainer,
  CardDetailsContainer,
  CardImage,
  CardDetailsMock,
  CardDetailsTitle,
} from './Card.styles';

const Card = ({ name, imageUrl, onClick }) => {
  return (
    <CardContainer onClick={onClick}>
      <CardImage alt="user" src={imageUrl} />
      <CardDetailsMock>
        <CardDetailsContainer>
          <CardDetailsTitle>{name}</CardDetailsTitle>
        </CardDetailsContainer>
      </CardDetailsMock>
    </CardContainer>
  );
};

Card.defaultProps = {
  onClick: () => {},
};

Card.propTypes = {
  name: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

export default Card;
