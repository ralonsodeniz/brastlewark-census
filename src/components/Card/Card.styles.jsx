import styled, { css } from 'styled-components';
// import { LazyImage } from 'react-lazy-images';
import { LazyLoadImage } from 'react-lazy-load-image-component';

import mediaQueryHelper from '../../style/media-queries';
import * as variables from '../../style/variables';

export const CardContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  background-color: transparent;
  cursor: pointer;
  -moz-osx-font-smoothing: grayscale;
  backface-visibility: hidden;
  width: 250px;
  height: 250px;
  box-shadow: 4px 4px 8px 0px rgba(0, 0, 0, 0.2);
  border-radius: 5px;

  ${mediaQueryHelper(
    'tab-port',
    css`
      width: 200px;
      height: 200px;
    `
  )};

  ${mediaQueryHelper(
    'phone',
    css`
      width: 125px;
      height: 125px;
    `
  )}

  & .placeholder {
    border-radius: 5px;
  }
`;

export const CardImage = styled(LazyLoadImage)`
  object-fit: cover;
  object-position: 50% 50%;
  height: 250px;
  width: 250px;
  border-radius: 5px;
  ${mediaQueryHelper(
    'tab-port',
    css`
      height: 200px;
      width: 200px;
    `
  )};
  ${mediaQueryHelper(
    'phone',
    css`
      height: 125px;
      width: 125px;
    `
  )};
`;

export const CardDetailsMock = styled.div`
  height: 40px;
  position: relative;
  display: flex;
  justify-content: center;

  ${mediaQueryHelper(
    'tab-port',
    css`
      height: 30px;
    `
  )};

  ${mediaQueryHelper(
    'phone',
    css`
      height: 20px;
    `
  )};
`;

export const CardDetailsContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: transparent;
  background-image: -webkit-linear-gradient(
    135deg,
    ${variables.primaryColor},
    ${variables.secondaryColor}
  );
  background-image: linear-gradient(135deg, ${variables.primaryColor}, ${variables.secondaryColor});
  -webkit-clip-path: polygon(50% 0, 100% 25%, 100% 100%, 0 100%, 0 25%);
  clip-path: polygon(50% 0, 100% 25%, 100% 100%, 0 100%, 0 25%);
  position: absolute;
  width: 100%;
  border-radius: 5px;
  height: 80px;
  bottom: 0px;
  text-align: center;
  color: ${variables.darkBlue};

  ${mediaQueryHelper(
    'tab-port',
    css`
      height: 60px;
    `
  )};

  ${mediaQueryHelper(
    'phone',
    css`
      height: 40px;
    `
  )};
`;

export const CardDetailsTitle = styled.h3`
  padding-top: 20px;
  font-weight: bold;
  font-size: ${variables.midFontSize};

  ${mediaQueryHelper(
    'tab-port',
    css`
      padding-top: 15px;
      font-size: ${variables.defaultFontSize};
    `
  )};

  ${mediaQueryHelper(
    'phone',
    css`
      padding-top: 10px;
      font-size: ${variables.smallFontSize};
    `
  )};
`;
