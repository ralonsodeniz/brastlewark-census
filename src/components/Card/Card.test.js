import React from 'react';
import { shallow, render } from 'enzyme';

import Card from './Card';
import { CardContainer } from './Card.styles';

describe('Card Component tests', () => {
  let wrapper;
  const mockCallBack = jest.fn();

  beforeEach(() => {
    const mockProps = {
      name: '',
      imageUrl: '',
      height: 0,
      onClick: mockCallBack,
    };

    wrapper = shallow(<Card {...mockProps} />);
  });

  test('expect Card to match shallow snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('expect CardContainer to match render snapshot to check styles', () => {
    expect(render(<CardContainer />)).toMatchSnapshot();
  });

  test('expect Card to executes on click the function passed as props', () => {
    wrapper.find(CardContainer).simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
