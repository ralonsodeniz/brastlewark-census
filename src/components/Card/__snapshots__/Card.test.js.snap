// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component tests expect Card to match shallow snapshot 1`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Styled(t)
    alt="user"
    height={0}
    placeholderSrc="test-file-stub"
    src=""
    wrapperClassName="placeholder"
  />
  <styled.div>
    <styled.div>
      <styled.h3 />
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Card Component tests expect CardContainer to match render snapshot to check styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  background-color: transparent;
  cursor: pointer;
  -moz-osx-font-smoothing: grayscale;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  width: 250px;
  height: 250px;
  box-shadow: 4px 4px 8px 0px rgba(0,0,0,0.2);
  border-radius: 5px;
}

.c0 .placeholder {
  border-radius: 5px;
}

@media screen and (max-width:56.25em) {
  .c0 {
    width: 200px;
    height: 200px;
  }
}

@media screen and (max-width:37.5em) {
  .c0 {
    width: 125px;
    height: 125px;
  }
}

<div
  class="c0"
/>
`;
